core/eolearn/tests/test_utils/test_parsing.py

```diff
@@ -0,0 +1,142 @@
+from dataclasses import dataclass
+from typing import List, Tuple, Union
+
+import numpy as np
+import pytest
+
+from eolearn.core import EOPatch, FeatureParser, FeatureType
+from eolearn.core.utils.parsing import FeatureRenameSpec, FeatureSpec, FeaturesSpecification
+from eolearn.core.utils.types import EllipsisType
+
+
+@dataclass
+class ParsingTestClass:
+    input: FeaturesSpecification
+    output_get_features: List[FeatureSpec]
+    output_get_renamed_features: List[FeatureRenameSpec]
+    output_get_feature_specifications: List[Tuple[FeatureType, Union[str, EllipsisType]]]
+
+
+TEST_CASES_EMPTY = [
+    ParsingTestClass(
+        input=(), output_get_features=[], output_get_renamed_features=[], output_get_feature_specifications=[]
+    ),
+]
+
+
+TEST_CASES = [
+    ParsingTestClass(
+        input=((FeatureType.DATA, "bands", "new_bands"),),
+        output_get_features=[(FeatureType.DATA, "bands")],
+        output_get_renamed_features=[(FeatureType.DATA, "bands", "new_bands")],
+        output_get_feature_specifications=[(FeatureType.DATA, "bands")],
+    ),
+    ParsingTestClass(
+        input=([(FeatureType.DATA, "bands"), (FeatureType.MASK, "bands", "new_bands")]),
+        output_get_features=[(FeatureType.DATA, "bands"), (FeatureType.MASK, "bands")],
+        output_get_renamed_features=[(FeatureType.DATA, "bands", "bands"), (FeatureType.MASK, "bands", "new_bands")],
+        output_get_feature_specifications=[(FeatureType.DATA, "bands"), (FeatureType.MASK, "bands")],
+    ),
+    ParsingTestClass(
+        input=(
+            {
+                FeatureType.DATA: [("bands", "new_bands"), ("CLP", "new_CLP")],
+                FeatureType.MASK: [("bands", "new_bands"), ("CLP", "new_CLP")],
```

Comment:
use something else rather than "bands" here to make it easier to find issues. bands -> IS_VALID, CLP -> CLM

Reply:
done

Reply:
not done, you still have (FeatureType.MASK, "bands") in multiple places even though it makes no sense (and neither does (FeatureType.DATA, "IS_VALID")

Reply:
done